aspnetcore/data/ef-rp/update-related-data.md

```diff
@@ -13,6 +13,221 @@ By [Tom Dykstra](https://github.com/tdykstra), and [Rick Anderson](https://twitt
 
 [!INCLUDE [about the series](../../includes/RP-EF/intro.md)]
 
+::: moniker range=">= aspnetcore-3.0"
+
+This tutorial shows how to update related data. The following illustrations show some of the completed pages.
+
+![Course Edit page](update-related-data/_static/course-edit30.png)
+![Instructor Edit page](update-related-data/_static/instructor-edit-courses30.png)
+
+## Update the Course Create and Edit pages
+
+The scaffolded code for the Course Create and Edit pages has a Department drop-down list that shows Department ID (an integer). The drop-down should show the Department name, so both of these pages need a list of department names. To provide that list, use a base class for the Create and Edit pages.
+
+### Create a base class for Course Create and Edit
+
+Create a *Pages/Courses/DepartmentNamePageModel.cs* file with the following code:
+
+[!code-csharp[](intro/samples/cu30/Pages/Courses/DepartmentNamePageModel.cs)]
+
+The preceding code creates a [SelectList](/dotnet/api/microsoft.aspnetcore.mvc.rendering.selectlist?view=aspnetcore-2.0) to contain the list of department names. If `selectedDepartment` is specified, that department is selected in the `SelectList`.
+
+The Create and Edit page model classes will derive from `DepartmentNamePageModel`.
+
+### Update the Course Create page model
+
+A Course is assigned to a Department. The base class for the Create and Edit pages provides a `SelectList` for selecting the department. The drop-down list that uses the `SelectList` sets the `Course.DepartmentID` foreign key (FK) property. EF Core uses the `Course.DepartmentID` FK to load the `Department` navigation property.
+
+![Create course](update-related-data/_static/ddl30.png)
+
+Replace the code in *Pages/Courses/Create.cshtml.cs* with the following code:
```

Comment:
Update *Pages/Courses/Create.cshtml.cs* with the following code:

Reply:
Do that everywhere you have "Replace the code in ... with the following code* 
too many codes sounds bad